clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/deploy/GCEUtil.groovy

```diff
@@ -188,6 +190,30 @@ class GCEUtil {
     }
   }
 
+  static def queryHealthCheck(String projectName, String healthCheckName, Compute compute, Task task, String phase) {
+    task.updateStatus phase, "Looking up http(s) health check $healthCheckName..."
+
+    try {
+      return compute.httpHealthChecks().get(projectName, healthCheckName).execute()
+    } catch (GoogleJsonResponseException e) {
+      // 404 is thrown, and the details are populated, if the health check does not exist in the given project.
+      // Any other exception should be propagated directly.
+      if (e.getStatusCode() == 404 && e.details) {
+        try {
+          return compute.httpsHealthChecks().get(projectName, healthCheckName).execute()
```

Comment:
Not flakiness. One's HTTP, one's HTTPS.
On Jul 4, 2016 8:53 AM, "Lars Wander" <notifications@github.com> wrote:

> In
> clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/deploy/GCEUtil.groovy
> <https://github.com/spinnaker/clouddriver/pull/732#discussion_r69453530>:
>
> > @@ -188,6 +190,30 @@ class GCEUtil {
> >      }
> >    }
> >
> > +  static def queryHealthCheck(String projectName, String healthCheckName, Compute compute, Task task, String phase) {
> > +    task.updateStatus phase, "Looking up http(s) health check $healthCheckName..."
> > +
> > +    try {
> > +      return compute.httpHealthChecks().get(projectName, healthCheckName).execute()
> > +    } catch (GoogleJsonResponseException e) {
> > +      // 404 is thrown, and the details are populated, if the health check does not exist in the given project.
> > +      // Any other exception should be propagated directly.
> > +      if (e.getStatusCode() == 404 && e.details) {
> > +        try {
> > +          return compute.httpsHealthChecks().get(projectName, healthCheckName).execute()
>
> If you're checking for flakiness here, should we use some sort of
> backoff/sleep before making a second identical request?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/spinnaker/clouddriver/pull/732/files/66aabfa4f853bd77189a3bcad54f8d853635e9a1#r69453530>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AIC_1r1_11pzbaD7L60WsViArn6V58hYks5qSQI9gaJpZM4JD45Y>
> .
>